[Liu Wenyi] ip

lwenyi1 opened the pull request.

lwenyi1 committed:
Adds *.class to gitignore

lwenyi1 committed:
Replaces Duke with Chloe in README

lwenyi1 committed:
Replaces Duke class with Chloe class

lwenyi1 committed:
Implements echo

lwenyi1 committed:
improves coding standard

lwenyi1 committed:
implements addTask() method

echo the text and add it to a taskList

lwenyi1 committed:
implements taskList() method

adds method to list tasks in order added

lwenyi1 committed:
adds the Task class

lwenyi1 committed:
updates Chloe methods to use Task class

lwenyi1 committed:
implements marking and unmarking

lwenyi1 committed:
Updates command entry system

The switch statement checks for bye, list, todo, deadline, event, mask and unmask now.
Also moved the task addition printing to the Task class constructor.

lwenyi1 committed:
Adds Deadline class

Adds and implements Deadline class, updates Chloe and Task classes to accomodate

lwenyi1 committed:
Adds Event class

Adds and implements the Event class.
Edits Chloe to accomodate changes.

kinneo commented on src/main/java/Chloe.java:
```diff
@@ -0,0 +1,190 @@
+ * greets the user, responds to commands, and says goodbye upon exiting.
+ * The program runs until the user types "bye".
+ */
+public class Chloe {
```
testing 1 2 3

kinneo reviewed (commented):
cool

mohammedhamdhan commented on src/main/java/Chloe.java:
```diff
@@ -0,0 +1,190 @@
+                listTasks();
+                break;
+            case "todo":
+                addTask(entryRemainder);
```
A suggestion would be to improve the name of "entryRemainder" to "taskToDo", or that would be more intuitive :)

mohammedhamdhan commented on src/main/java/Chloe.java:
```diff
@@ -0,0 +1,190 @@
+    /**
+     * Prints message asking user for valid command.
+     * */
+    public static void handleBadCommand() {
+        System.out.println(LINE);
+        System.out.println("\tI'm sorry I don't know what you mean!!");
+        System.out.println("\tEnter a valid command please :\")");
+        System.out.println(LINE);
+    }
+
+    /**
+     * Reads in task from the user and adds it to the task list.
+     * */
+    public static void addTask(String task) {
+        // Adds task to the list
+        Task newTask = new Task(task);
+        newTask.printTaskAddition();
+        taskList.add(newTask);
+    }
+
+    /**
+     * Reads in deadline from the user and adds it to the task list.
+     * */
+    public static void addDeadline(String task) {
+        // Split task name and due date
+        String[] stringParts = task.split(" /by ", 2);
+        String taskDescription = stringParts[0];
+        String dueDate = stringParts.length > 1? stringParts[1] : "";
+
+        // Adds deadline to the list
+        Deadline newDeadline = new Deadline(taskDescription, dueDate);
+        newDeadline.printTaskAddition();
+        taskList.add(newDeadline);
+    }
```
A suggestion would be to make a new class, eg: TaskManager, and use that class to handle all adding/marking/unmarking of tasks. This would make your code more modular :)

mohammedhamdhan commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,31 @@
+        super(taskDescription);
+        this.dueDate = dueDate;
+    }
+
+    @Override
+    public void printTaskAddition() {
+        // Echo the command back to the user
+        System.out.println(LINE);
+        System.out.println("\tAdded \"" + getTaskDescription() + "\" for you <3, do it by: "
+                + getDueDate() +"!!");
+        System.out.println(LINE);
+    }
+
+    @Override
+    public String getTaskType() {
+        return "[D]";
+    }
+
+    public String getDueDate() {
+        return dueDate;
+    }
+
+    @Override
```
good use of Override! It makes your code easy to read and follow!

mohammedhamdhan commented on src/main/java/Chloe.java:
```diff
@@ -0,0 +1,190 @@
+    /**
+     * Prints message asking user for valid command.
+     * */
+    public static void handleBadCommand() {
```
This is a good way of handling bad user commands!

lwenyi1 committed:
Improves readability of code based on PR comments

lwenyi1 committed:
Implements testing script

lwenyi1 committed:
Adds error handling

lwenyi1 committed:
Implements TaskManager class

Creates a class to manage the tasks for Chloe

lwenyi1 committed:
Implements exception throwing for illegal and incomplete commands

lwenyi1 committed:
Merge branch 'branch-Level-5'

lwenyi1 committed:
Moves files into chloe package

lwenyi1 committed:
Moves Task, Deadline and Event classes into tasktypes package

lwenyi1 committed:
Moves exception classes into exception package.

lwenyi1 committed:
Merge branch 'branch-A-Packages'

lwenyi1 committed:
Updates README to describe todo, deadline and event features.

lwenyi1 committed:
Updates README to describe mark and unmark features.

ljunfun commented on src/main/java/chloe/Chloe.java:
```diff
@@ -1,7 +1,9 @@
+package chloe;
+
 import java.util.*;
```
Bad coding practice. Imported classes should always be listed explicitly

ljunfun commented on src/main/java/chloe/TaskManager.java:
```diff
@@ -1,5 +1,10 @@
+package chloe;
+
 import java.util.*;
```
Imported classes should always be listed explicitly

lwenyi1 replied:
ok

lwenyi1 committed:
Fixes coding style by listing packages explicitly

lwenyi1 committed:
Implements handleBadIndex()

lwenyi1 committed:
Implements task deletion

lwenyi1 committed:
Updates test code for task deletion

lwenyi1 committed:
Implements save file

lwenyi1 committed:
Merge branch 'branch-Level-6'

lwenyi1 committed:
Merge branch 'branch-Level-7'

lwenyi1 committed:
Updates testing process to check for file saving

lwenyi1 committed:
Adds removal of temp files to test script

lwenyi1 committed:
Creates JAR file (find it in `releases`)

lwenyi1 committed:
Updates documentation for task saving process

aditichadha1310 commented on src/main/java/chloe/TaskManager.java:
```diff
@@ -0,0 +1,306 @@
+                System.out.println("Error creating tasks file: " + e.getMessage());
+            }
+        }
+
```
Avoid arrow head style code.

aditichadha1310 commented on src/main/java/chloe/TaskManager.java:
```diff
@@ -0,0 +1,306 @@
+        try (BufferedReader reader = new BufferedReader(new InputStreamReader(new FileInputStream(file), StandardCharsets.UTF_8))) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                String[] parts = line.split(" # "); // Use # to split task components
```
Avoid usage of magic strings.

aditichadha1310 reviewed (commented):
overall, good job!

lwenyi1 committed:
Abstracts storage class

Creates storage class and UI class files, abstracts storage class but leaves UI class as an empty placeholder file

lwenyi1 committed:
Creates UI, TaskList and Parser classes

lwenyi1 committed:
Creates command package

Implements AddTodo and AddDeadline classes along with it

lwenyi1 committed:
Implements AddEvent and ListTask classes

lwenyi1 committed:
Implemetns DeleteTask and MarkCommand

lwenyi1 committed:
Improves abstraction in UI

lwenyi1 committed:
Implements saving as LocalDateTime type

Updates AddDeadline, AddEvent, Deadline, Event and the test file etc. to accomodate

lwenyi1 committed:
Implements find feature

lwenyi1 committed:
Improves documentation for find task

lwenyi1 committed:
Adds javadoc documentation

lwenyi1 committed:
Merge pull request #1 from lwenyi1/branch-Level-8

Level-8

lwenyi1 committed:
Merge branch 'master' into branch-Level-9

lwenyi1 committed:
Merges pull request #2 from lwenyi1/branch-Level-9

Level-9

lwenyi1 committed:
Merge branch 'master' into branch-A-JavaDoc

lwenyi1 committed:
Merges pull request #3 from lwenyi1/branch-A-JavaDoc

A-JavaDoc

lwenyi1 committed:
Merge branch 'master' of https://github.com/lwenyi1/ip

lwenyi1 committed:
Updates user guide

lwenyi1 committed:
Fixes some date/time bugs

lwenyi1 committed:
Updates user guide
